src/machine/machinectl.c

```diff
@@ -505,18 +558,34 @@ static int print_addresses(sd_bus *bus, const char *name, int ifi, const char *p
                 if (r < 0)
                         return bus_log_parse_error(r);
 
-                fputs(prefix, stdout);
-                fputs(inet_ntop(family, a, buffer, sizeof(buffer)), stdout);
-                if (family == AF_INET6 && ifi > 0)
-                        printf("%%%i", ifi);
-                fputc('\n', stdout);
+                if (n_addr != 0) {
+                        if (family == AF_INET6 && ifi > 0) {
+                                r = sprintf(buf_ifi, "%%%i", ifi);
```

Comment:
if you are not interested in the rturn value, there's no need to assign it to "r" here...

Consider using xsprintf() in this case.